rstudio-tidyverse/rockcraft.yaml

```
@@ -0,0 +1,259 @@
+# Based on:
+# https://github.com/kubeflow/kubeflow/blob/v1.10.0-rc.1/components/example-notebook-servers/rstudio-tidyverse/Dockerfile
+# https://github.com/kubeflow/kubeflow/blob/v1.10.0-rc.1/components/example-notebook-servers/rstudio/Dockerfile
+# https://github.com/kubeflow/kubeflow/blob/v1.10.0-rc.1/components/example-notebook-servers/base/Dockerfile
+
+name: rstudio-tidyverse
+summary: An image for using tidyverse
+description: |
+  The tidyverse is an opinionated collection of R packages designed for data science. 
+  Tidyverse are installed in Conda environment, which is automatically activated.
+version: "v1.10.0-rc.1"
+license: Apache-2.0
+base: ubuntu@22.04
+run-user: _daemon_
+
+platforms:
+  amd64:
+
+services:
+  rstudio-tidyverse:
+    override: replace
+    summary: "Rstudio Tidyverse service"
+    startup: enabled
+    command: /usr/lib/rstudio-server/bin/rserver --server-daemonize=0 --server-data-dir=/tmp/rstudio_runtime --server-working-dir=/home/_daemon_ --server-user=_daemon_ --rsession-which-r=/opt/conda/lib/R --www-address=0.0.0.0 --www-port=8888 --auth-none=1 --rsession-path=/opt/rsession.sh --rsession-ld-library-path=/opt/conda/lib
+    working-dir: /home/
+    environment:
+       NB_USER: _daemon_
```

suggestion(blocking): To be more in line with the other rocks, let's have the notebook user be `joyvan` here as well
```suggestion
       NB_USER: _daemon_
```

---

If we start service with jovyan as user in this rock we gonna get error: `Change 1 task (Start service "rstudio-tidyverse") failed: cannot start service: fork/exec /usr/lib/rstudio-server/bin/rserver: operation not permitted` . I asked about this in matrix chat and Kimonas Sotirchos answered me with:
``` 
you can't use run-user: _daemon_ and also services.rstudio-tidyverse.user at the same time
if you set run-user this means that pebble (init system inside the rock) will be run as a non-root user
then if you have services.rstudio-tidyverse.user at the same time, you are asking pebble to launch the rstudio process with a different user
but your non-root pebble process doesn't have permissions to spin up other processes with a different uid, since it's not root
my suggestion is:
1. Use only run-user: _daemon_
2. Expect that the user should not be jovyan buth _daemon_
  1.This might require doing chown in places that are explicitly owned by jovyan
from what I remember the jovyan name isn't hardcoded, in the sense that we could have a different user. So the above plan should work
```
so i remade rock to use daemon instead of jovyan

---

You are right, we are unable to run Pebble as `_daemon_` and the service as `jovyan`. We can do one of the following:
- Change every reference from `jovyan` to `daemon`, as you do
- Keep all files under the `/home/jovyan` directory as the upstream Dockerfile does, and change the permissions to allow access from the `_daemon_` user.

I would suggest that we go with the second option, since this is what we do with the other rocks, see for example [jupyter-scipy](https://github.com/canonical/kubeflow-rocks/blob/main/jupyter-scipy/rockcraft.yaml)